xbmc/cores/FFmpeg.cpp

```diff
@@ -135,15 +135,78 @@ void ff_avutil_log(void* ptr, int level, const char* format, va_list va)
   buffer.erase(0, start);
 }
 
-std::tuple<uint8_t*, int> GetPacketExtradata(const AVPacket* pkt, const AVCodecParameters* codecPar)
+FFmpegExtraData::FFmpegExtraData(size_t size)
+  : m_data(reinterpret_cast<uint8_t*>(av_malloc(size + AV_INPUT_BUFFER_PADDING_SIZE))), m_size(size)
```

Comment:
We seem to use a mixture of av_malloc and av_mallocz. I'm not sure really if we need to zero out.

```suggestion
  : m_data(reinterpret_cast<uint8_t*>(av_mallocz(size + AV_INPUT_BUFFER_PADDING_SIZE))), m_size(size)
```

Reply:
I lean towards not to zero out by default because the user has to fill the buffer with data in any case. If the user doesn't initialize the buffer it probably is a bug and zeroing the buffer could hide that from tools like Valgrind.